rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -105,8 +108,17 @@ def put(self, deployment_id, **kwargs):
         if not args.async_create:
             sm = get_storage_manager()
             rest_utils.wait_for_execution(sm, deployment.create_execution.id)
+            if deployment.create_execution.status != ExecutionState.TERMINATED:
+                raise self._error_from_create(deployment.create_execution)
         return deployment, 201
 
+    def _error_from_create(self, execution):
+        """Map a failed create-dep-env execution to a REST error response"""
+        if execution.status != ExecutionState.FAILED or not execution.error:
+            return DeploymentEnvironmentCreationInProgressError()
+        error_message = execution.error.strip().split('\n')[-1]
+        return DeploymentCreationError(error_message)
```

Comment:
for now just this, but in the future we'll look at the error and return some of the specialized error classes, eg. `UnknownDeploymentInputError` et al